node_modules/oae-ui/lib/api.js

```diff
@@ -811,6 +813,37 @@ var _replaceOptimizedPaths = function(skinVariables) {
     return replacedVariables;
 };
 
+/**
+ * Uploads a logo file for a tenant.
+ *
+ * @param  {Context}    ctx             Current request context
+ * @param  {Object}     file            A file object as returned by express
+ * @param  {String}     tenantAlias     The alias of the tenant for which to upload the logo
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ * @param  {String}     callback.url    The signed URL for the new logo file
+ */
+var uploadLogoFile = module.exports.uploadLogoFile = function(ctx, file, tenantAlias, callback) {
+    // Need to require this here to avoid circular dependency issues
+    var ContentAPI = require('oae-content/lib/api');
+    tenantAlias = tenantAlias || ctx.tenant().alias;
+    var tenant = ctx.tenant();
+    if (!ctx.user() || !ctx.user().isAdmin(tenantAlias)) {
+        return callback({'code': 401, 'msg': 'Only administrators can upload new logos for tenants'});
+    }
+    ContentAPI.createFile(ctx, util.format('%s logo', tenantAlias), util.format('Logo displayed on %s landing page', tenantAlias), AuthzConstants.visibility.PRIVATE, file, {}, [], function(err, content) {
```

Comment:
Maybe we should change this too:
```javascript
ContentUtil.getStorageBackend(ctx).store(ctx.tenant().alias, file, options, function(err, uri) {
```

That way the logo won't end up in the search index or the admin's content library.